app/faceanalysis/models/database_manager.py

```diff
@@ -53,3 +40,8 @@ def safe_commit(self, session):
     def close_engine(self):
         self.logger.debug("engine closed")
         self.engine.dispose()
+
+
+@lru_cache(maxsize=1)
```

Comment:
Is the purpose of this decorator (maxsize=1, on a function with no arguments) to treat the DatabaseManager class as a singleton? If so, is this a better way than explicitly making the class a singleton, so it is not accidentally misused?

Reply:
Yes, this is the standard/stdlib way to create a singleton.

Reply:
The problem I foresee with this method is that I could still make as many different instantiations of DatabaseManager as I would like. If we made the class an actual singleton (https://www.python-course.eu/python3_metaclasses.php) (could use metaclasses as the link describes, but not totally necessary), then this problem would be alleviated.

Reply:
In dynamic languages you can always shoot yourself in the foot if you want to... it's just about making "doing the right thing" easy. Ideally the design would be changed to avoid the need for the singleton (e.g. currently there are lots of nested sessions etc flying around) but that's a bigger change than I'm willing to make right now.